### ISSUE TITLE
Warning: Tried to initialize from a previous compilation (build/app.dill.track.dill), but the file was not self-contained. This might be a bug.

### ISSUE THREAD

**Issue by pjdevries** [dart-lang/sdk#36770]
Thank you for taking the time to file an issue!

In order to route, prioritize, and act on this, please include:

* Dart SDK Version (`dart --version`)
Dart VM version: 2.2.0 (Tue Feb 26 15:04:32 2019 +0100) on "windows_x64"

* Whether you are using Windows, MacOSX, or Linux (if applicable)
Windows 10

* Whether you are using Chrome, Safari, Firefox, Edge (if applicable)

Flutter 1.5.9-pre.28 • channel master • https://github.com/flutter/flutter.git
Framework • revision 14f2f98f9d (2 hours ago) • 2019-04-26 07:55:29 -0700
Engine • revision 0b6a4be5f1
Tools • Dart 2.3.0 (build 2.3.0-dev.0.3 7adad2a245)

Android Studio 3.4
Build #AI-183.5429.30.34.5452501, built on April 10, 2019
JRE: 1.8.0_152-release-1343-b01 amd64
JVM: OpenJDK 64-Bit Server VM by JetBrains s.r.o
Windows 10 10.0

Missing some or all of the above might make the issue take longer or be
impossible to act on.

----
[initialize_from.dill.gz](https://github.com/dart-lang/sdk/files/3122252/initialize_from.dill.gz)

**Comment by devoncarew** [comment]
> but the file was not self-contained

We have seen several similar reports; is this an issue, or should we not be asking for users to report this?

@jensjoha

**Comment by pjdevries** [comment]
I can not answer that question. I'm just an end user of Dart & Flutter, not at all familiar with the inner workings or specific technical aspects of the programming environment. The post was in a response to the explicit request indeed.

What I can say is that I have been working on a daily basis with the described setup for the past 6 or 7 weeks and this is the first time it happened to me.

Regards,
Pieter-jan de Vries

**Comment by jensjoha** [comment]
Thank you for the report.

This is the same issue as #36759 and I'll copy my response from there:

This is caused by the previous compilation being compiled on the basis of an older version of the sdk where the new version (the one now included in flutter) has changed (specifically in flutter/engine@3e47b4b the getter "left" on class "Rect" was removed and became a field instead.)

This means that the old compilation tries to reference something that no longer exists. This is detected, and we're saying that we cannot initialize from it. This is perfectly fine. Ideally we wouldn't even try to initialize from it, because we would know that the sdk versions doesn't match up, but we cannot do that (yet). #36032 tracks that.

My 2 cents is, that the message is good to have. Once in a while we'll get a few messages reporting this. This will continue to happen until #36032 is fixed.
If - both before and after #36032 has been fixed - we suddenly get very many (or, for that matter, that upon investigating, we'll find that something has indeed broken with loading from dill) we will know much sooner (and can thus fix it much sooner) than if we didn't report it.

In the past we've had the loading from dill feature not working at all (because the file we tried to load from was overwritten with a partial file) without any kind of warning at all. This would go unnoticed until someone realized that compilation was slower than it used to be and started investigating. Clearly that's bad, and in my opinion worse than getting a few "false positives" once in a while.

Again, @pjdevries, thank you very much for taking the time to report this issue. It is very much appreciated!
